File: Python/insertion_sort.py
A = [5, 10, 5, 2, 5, 9, 2, 6, 5, 3, 2]
print("Input:", A)
ops = 0
for k in range(1, len(A)):
    item = A[k]
    i = k

    while i > 0 and item < A[i - 1]:
        A[i] = A[i-1]
        i -= 1
        ops += 1
    A[i] = item
    # print(A)

print("No of Operations: ", ops)
print("Output:", A)

File: C++/insertion_sort.cpp
#include <iostream>
#include <vector>

using namespace std;

int insertionSort(vector<int>& arr) {
    int ops = 0;
    for(int idx = 1; idx <size(arr); ++idx) {
        int item = arr[idx];
        while (idx > 0 && item < arr[idx-1])
        {
            arr[idx] = arr[idx - 1];
            --idx;
            ++ops;
        }
        arr[idx] = item;
    }

    return ops;
}

int main() {
    vector<int> arr = {5, 12, 3, 4, 7, 1, 0, 6, 19, 8, 13, 4, 2, 10, 16};

    cout << "input: ";
    for(int num: arr){
        cout << num << " ";
    }    
    cout << endl;

    int ops = insertionSort(arr);
    cout << "No of operations: " << ops << endl;

    cout << "Output: ";
    for (int i = 0; i < arr.size(); ++i) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}